runtime/compiler/z/codegen/S390PrivateLinkage.cpp

```diff
@@ -1526,6 +1531,238 @@ J9::Z::PrivateLinkage::createEpilogue(TR::Instruction * cursor)
+   if (numITableIterationsAfterLastITableCacheCheck)
+      {
+      iterations = atoi(numITableIterationsAfterLastITableCacheCheck);
+      if (trace)
+         traceMsg(comp, "ITable iteration is %d because TR_NumITableIterationsAfterLastITableCacheCheck was set.\n", iterations);
+      return iterations;
+      }
+
+   if (!comp->getOption(TR_DisableCHOpts))
+      {
+      TR_ResolvedMethod **implArray = new (comp->trStackMemory()) TR_ResolvedMethod*[MAX_IMPLEMENTERS_TO_EVALUATE+1];
+      TR_PersistentCHTable *chTable = comp->getPersistentInfo()->getPersistentCHTable();
+      int32_t cpIndex = methodSymRef->getCPIndex();
+      // Find out how many implementers are for the declaring interface class.
+      int32_t numImplementers = chTable->findnInterfaceImplementers(declaringClass, MAX_IMPLEMENTERS_TO_EVALUATE, implArray, cpIndex, methodSymRef->getOwningMethod(comp), comp);
+      if ((numImplementers != 0) && (numImplementers <= MAX_IMPLEMENTERS_TO_EVALUATE))
+         {
+         // Find out how many interfaces each implementer implements
+         int32_t maxInterfaces = MIN_ITABLE_ITERATIONS;
+
+         for (int32_t i=0; i < numImplementers; ++i)
+            {
+            TR_OpaqueClassBlock *containingClass = implArray[i]->containingClass();
+            int32_t numInterfaces = fej9->numInterfacesImplemented((J9Class *)containingClass);
+            maxInterfaces = (numInterfaces > maxInterfaces) ? numInterfaces : maxInterfaces;
+            }
+
+         iterations = (maxInterfaces > MAX_ITABLE_ITERATIONS) ? MAX_ITABLE_ITERATIONS : maxInterfaces;
+
+         if (trace)
+            traceMsg(comp, "ITable declaringClass %p numImplementers %d maxInterfaces %d iterations %d\n", declaringClass, numImplementers, maxInterfaces, iterations);
+         }
+      }
+   else if (trace)
+      traceMsg(comp, "ITable iteration is the default value of %d.\n", iterations);
+   return iterations;
+   }
+
+TR::Instruction *
+generateLastITableAndITableInstructions(TR::CodeGenerator * cg, TR::Node * callNode, TR::Register * vftReg,
+   TR::Register * entryPointRegister, TR::Register * vTableIndexRegister, TR_S390ScratchRegisterManager *srm, TR::Instruction * cursor)
+   {
+   TR::Compilation * comp = cg->comp();
+   TR_J9VMBase *fej9 = reinterpret_cast<TR_J9VMBase *>(comp->fe());
+   uintptr_t itableIndex;
+   TR_OpaqueClassBlock *declaringClass = NULL;
+   TR_Debug * debugObj = cg->getDebug();
+   TR::SymbolReference * methodSymRef = callNode->getSymbolReference();
+
+   // Set the default state of enableLastITableCache.
+   bool enableLastITableCache = !cg->profiledPointersRequireRelocation() && !comp->getOption(TR_DisableLastITableCache);
+
+   static char* forceInterfaceDispatchUsingLastITable = feGetEnv("TR_forceInterfaceDispatchUsingLastITableState");
+   if (forceInterfaceDispatchUsingLastITable)
+      enableLastITableCache = static_cast<bool>(atoi(forceInterfaceDispatchUsingLastITable));
```

Comment:
Do you intend to provide this option to force generation of ITable Cache / walk when we are generating an AOT code? I think we should be able to simplify this bit of the code. Also if the intention of the env to just use as `true` or `false` we should avoid calling `atoi` un-necessary. Simply `feGetEnv(TR...) != NULL` is sufficient.

Also if you are not planning to later-on use this flag to enable it for AOT, I would recommend removing it.